src/services/arweave/SimpleArweaveDataProvider.ts

```diff
@@ -25,20 +30,18 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
   }
 
   async getTransactionStatus(id: ArweaveTransactionID) {
-    const confirmations = await this._arweave.api
-      .get(`/tx/${id}/status`)
-      .then((res: any) => {
-        return res.data.number_of_confirmations;
-      });
-    return confirmations;
+    const { status, data } = await this._arweave.api.get(`/tx/${id}/status`);
+    if (status !== 200) {
+      throw Error('Failed fetch confirmations for transaction id.');
+    }
+    return +data.number_of_confirmations;
   }
```

Comment:
the arweave client swallows all errors (404's, 429's etc.) so we want to to check the `status` and throw our own error